frontend/src/components/Sidepanel/Panels/TestingLab/TestingLab.tsx

```diff
@@ -52,6 +53,45 @@ const TestingLab = () => {
   const projectType = useProjectStore(s => s.project.config.type)
   const setPDAVisualiser = usePDAVisualiserStore(state => state.setStack)
 
+  // Add Stepper from (old) SteppingLab
+  const stepper = useMemo(() => {
+    // Graph stepper for PDA currently requires changes to BFS stack logic
+    // to handle non-determinism so branching stops on the first rejected transition.
+    return graphStepper(graph as FSAProjectGraph | PDAProjectGraph | TMProjectGraph, traceInput)
+  }, [graph, traceInput])
+  const setSteppedStates = useSteppingStore(s => s.setSteppedStates)
+
+  const handleStep = (stepType: StepType) => {
+    let frontier = []
+    if (stepper) {
+      switch (stepType) {
+        case 'Forward':
+          frontier = stepper.forward()
+          break
+        case 'Backward':
+          frontier = stepper.backward()
+          break
+        case 'Reset':
+          frontier = stepper.reset()
+          break
+        default:
+          break
+      }
+
+      if (frontier && frontier.length > 0) {
+        setSteppedStates(frontier)
+      }
+    }
+  }
+
+  useEffect(() => {
+    if (stepper) {
+      handleStep('Reset')
+    }
+  }, [traceInput])
+
+  const noStepper = stepper === null && false
```

Comment:
Apparently I wrote this line whoops
Its always false no? So can be safely removed

Reply:
Should be done now, thanks for picking that up. It's definitely always false.
Surely there's a reason why you've wrote that in the Stepping Lab... right? :)

Reply:
🤷‍♂️
Have a feeling it was debugging stuff I accidentally left in lol